src/argilla/client/feedback/dataset/remote.py

```diff
@@ -367,3 +369,17 @@ def add_records(
             ValueError: if the given records do not match the expected schema.
         """
         self._records.add(records=records, show_progress=show_progress)
+
+    @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
+    def delete(self) -> None:
+        """Deletes the current `FeedbackDataset` from Argilla. This method is just working
+        if the user has either `owner` or `admin` role.
+
+        Raises:
+            PermissionError: if the user does not have either `owner` or `admin` role.
+            Exception: if the `FeedbackDataset` cannot be deleted from Argilla.
+        """
+        try:
+            datasets_api_v1.delete_dataset(client=self._client, id=self.id)
+        except Exception as e:
+            raise Exception(f"Failed while deleting the `FeedbackDataaset` from Argilla with exception: {e}") from e
```

Comment:
I think that we should try to avoid using plain `Exception`s, maybe it's better to raise a `RuntimeError` exception here.

Reply:
I also agree, maybe I can create another PR for this (or the next) release to create some custom exceptions for the Python client

Reply:
For the moment I'll just change this to `RuntimeError` as you suggested!

Reply:
```suggestion
            raise RuntimeError(f"Failed while deleting the `FeedbackDataset` from Argilla with exception: {e}") from e
```